### ISSUE TITLE
The loss didn't decrease when training

### ISSUE THREAD

**Issue by RaymondLmx** [Nanne/pytorch-NetVlad#32]
Can you share some experiences about how to set learning rate, margin and threads for I used a totally different dataset. The loss keep steady in the training. I have tried adjusting the rate, margin but invalid.

**Comment by Nanne** [comment]
I would say that learning rate has more to do with the batch size than the dataset, and threads shouldn't affect the loss. Margin can be a tricky one to get right, as its a bit fiddly from what I've seen, but you could grid search it as its just a single parameter.

In my experience the netvlad loss can be quite tricky to get right, but there's a number of other losses which give good results as well, we had a paper last year that compares several: https://openaccess.thecvf.com/content_ICCVW_2019/papers/CVFAD/Ibrahimi_Deep_Metric_Learning_for_Cross-Domain_Fashion_Instance_Retrieval_ICCVW_2019_paper.pdf

If you've committed to using Netvlad then I'm affraid its just trial & error + grid searching..
